service/controller/resource/masters/create_restart_kubelets.go

```diff
@@ -0,0 +1,99 @@
+package masters
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/azure-operator/service/controller/controllercontext"
+	"github.com/giantswarm/azure-operator/service/controller/internal/state"
+	"github.com/giantswarm/azure-operator/service/controller/key"
+)
+
+func (r *Resource) restartKubeletOnWorkersTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	// Check if API server is up or wait.
+	r.logger.LogCtx(ctx, "level", "debug", "message", "Checking if API server is up")
+	up, err := r.isApiServerUP(ctx)
+	if err != nil || !up {
+		msg := "API server is NOT up."
+		if err != nil {
+			msg = fmt.Sprintf("%s Original error was %s", msg, err.Error())
+		}
+		r.logger.LogCtx(ctx, "level", "debug", "message", msg)
+
+		return currentState, nil
+	}
```

Comment:
I think combining checks here is unnecessary. It would be easier to read and probably shorter to write if they are checked separately :slightly_smiling_face: